File: Aplicacion/ClubDeViaje/ServicioClubDeViaje.cs
using Dominio.Dto;
using Dominio.Entity;
using Dominio.Interfaces.Aplicacion.ClubDeViaje;
using Dominio.Interfaces.Infraestrutura.BaseDatos;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Aplicacion.ClubDeViaje
{
    public class ServicioClubDeViaje : IServicioClubDeViaje
    {
        private readonly IRepositorioClubDeViaje repositorioClubDeViaje;

        public ServicioClubDeViaje(IRepositorioClubDeViaje repositorioClubDeViaje)
        {
            this.repositorioClubDeViaje = repositorioClubDeViaje;
        }

        public async Task<DtoDatosSP> ActualizarClubDeViaje(EntityClubDeViaje entityClubDeViaje)
        {
            return await this.repositorioClubDeViaje.ActualizarClubDeViaje(entityClubDeViaje);
        }

        public async Task<DtoDatosSP> EliminarClubDeViaje(int idClubDeViaje)
        {
            return await this.repositorioClubDeViaje.EliminarClubDeViaje(idClubDeViaje);
        }

        public async Task<DtoDatosSP> InsertarClubDeViaje(EntityClubDeViaje entityClubDeViaje)
        {
            return await this.repositorioClubDeViaje.InsertarClubDeViaje(entityClubDeViaje);
        }

        public async Task<DtoClubDeViaje> ObtenerDetalleClubDeViajeID(int idClubDeViaje)
        {
            return await this.repositorioClubDeViaje.ObtenerDetalleClubDeViajeID(idClubDeViaje);
        }

        public async Task<List<DtoClubDeViaje>> ObtenerClubDeViaje()
        {
            return await this.repositorioClubDeViaje.ObtenerClubDeViaje();
        }

        public async Task<List<EntityTipoCuenta>> ObtenerTiposDeCuentas()
        {
            return await this.repositorioClubDeViaje.ObtenerTiposDeCuentas();
        }

        public async Task<List<EntityTipoTarjeta>> ObtenerTiposDeTarjetas()
        {
            return await this.repositorioClubDeViaje.ObtenerTiposDeTarjetas();
        }

        public async Task<List<EntityFrecuenciaDePago>> ObtenerFrecuenciasDePago()
        {
            return await this.repositorioClubDeViaje.ObtenerFrecuenciasDePago();
        }
    }
}

File: WebUI/ClientApp/src/servicios/ServicioClubDeViaje.js
import { ProcesarDatosApi } from "../api/ApiFetch";
const controlador = "ClubDeViaje";

export const ObtenerClubDeViaje = async() => {
    const url = `${controlador}/ObtenerClubDeViaje`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerClubDeViajePorId = async(id) => {
    const url = `${controlador}/ObtenerDetalleClubDeViajeID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const InactivarClubDeViaje = async (id) => {
    const url = `${controlador}/EliminarClubDeViaje?idClubDeViaje=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const AgregarClubDeViaje = async (data) => {
    const url = `${controlador}/insertarClubDeViaje`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarClubDeViaje = async (data) => {
    const url = `${controlador}/actualizarClubDeViaje`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const ObtenerTiposDeCuentas = async() => {
    const url = `${controlador}/ObtenerTiposDeCuentas`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerTiposDeTarjetas = async() => {
    const url = `${controlador}/ObtenerTiposDeTarjetas`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerFrecuenciasDePago = async() => {
    const url = `${controlador}/ObtenerFrecuenciasDePago`;
    return await ProcesarDatosApi('GET', url);
}

